src/Frontend/Base.php

```diff
@@ -131,6 +133,80 @@ abstract public function add_checkout_tab( $tabs, $response );
 	 */
 	abstract public function get_content_data( $response, $post_data );
 
+	public function add_pickup_points_fields_to_email($fields, $sent_to_admin, $order) {
+		$data = $this->get_data( $order->get_id() );
+		if( ! empty( $data['frontend'] ) ){
+			$value = $this->generate_pickup_points_email_html($data['frontend']);
+			$fields['Pickup address'] = array(
+				'label' => __( 'Pick up at PostNL-point', 'postnl-for-woocommerce' ),
+				'value' => $value
+			);
+		}
+		return $fields;
+	}
+
+	/**
+	 * Generate the dropoff points html information.
+	 *
+	 * @param Array $infos Dropoff points informations.
+	 */
+	public function generate_pickup_points_email_html( $infos ) {
+		$filtered_infos = array_filter(
+			$infos,
+			function ( $info ) {
+				$displayed_info = array(
+					'dropoff_points_date',
+					'dropoff_points_time',
+				);
+
+				return in_array( $info, $displayed_info, true );
+			},
+			ARRAY_FILTER_USE_KEY
+		);
+
+		$address_info = array_filter(
+			$infos,
+			function ( $info ) {
+				return false !== strpos( $info, '_address_' );
+			},
+			ARRAY_FILTER_USE_KEY
+		);
+
+		if ( ! empty( $address_info ) ) {
+			$filtered_infos['address'] = implode( ', ', $address_info );
+			ksort( $filtered_infos );
+		}
+
+		if ( empty( $filtered_infos ) ) {
+			return;
+		}
```

Comment:
I can see this is a duplicated code (class Single.php), we can move it to Utils instead of duplicating it since it use the same thing in both classes